Tasks/NuGetInstaller/task.json

```diff
@@ -79,25 +79,30 @@
                 "Normal": "Normal",
                 "Detailed": "Detailed"
             }
+        },   
+        {
+            "name": "nuGetVersion",
+            "type": "radio",
+            "label": "NuGet Version",
+            "defaultValue": "3.3.0",
+            "helpMarkDown": "The version of NuGet to use, or external verson.",
+            "required": "true",
+            "groupName": "advanced",
+            "options": {
+                "3.3.0": "3.3.0",
+                "3.5.0.1829": "3.5.0 – build 1829 (rc2)",
```

Comment:
This is going to introduce a potential big compat problem.  That string 3.5.0.1829 value is going to get persisted in folks definitions.  You can't remove it (or at a minimum you can't remove runtime support for getting that value) going forward.

I would have done it as a combo box with a couple suggested versions with the ability to type in versions.  Then the installer would know how to go get nuget by any version from a public end point.

Then you're down to how long nuget versions stick around at a public endpoint which is good - that's an external policy decision.